src/main/java/org/broadinstitute/hellbender/utils/iterators/ReadTransformingIterator.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.hellbender.utils.iterators;
+
+import org.broadinstitute.hellbender.transformers.ReadTransformer;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * An iterator that transforms read (i.e. implements a function GATKRead -> GATKRead) from an existing iterator of reads.
+ * This is equivalent to a stream.map(readTransformer) operation.
+ */
+public class ReadTransformingIterator implements Iterator<GATKRead>, Iterable<GATKRead> {
```

Comment:
Add a basic unit test for `ReadTransformingIterator` with a dummy read transformer that performs some simple and easily-detected transformation, and assert that the transformation gets applied and you get the correct number of reads out of the iterator.